unity-renderer/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs

```diff
@@ -130,24 +136,24 @@ public static IEnumerator ApplyModelChanges(IParcelScene scene, TMP_Text text, M
 
             if (model.shadowOffsetX != 0 || model.shadowOffsetY != 0)
             {
-                text.fontMaterial.EnableKeyword("UNDERLAY_ON");
-                text.fontMaterial.SetColor("_UnderlayColor", model.shadowColor);
-                text.fontMaterial.SetFloat("_UnderlaySoftness", model.shadowBlur);
+                text.fontSharedMaterial.EnableKeyword("UNDERLAY_ON");
+                text.fontSharedMaterial.SetColor("_UnderlayColor", model.shadowColor);
+                text.fontSharedMaterial.SetFloat("_UnderlaySoftness", model.shadowBlur);
             }
-            else if (text.fontMaterial.IsKeywordEnabled("UNDERLAY_ON"))
+            else if (text.fontSharedMaterial.IsKeywordEnabled("UNDERLAY_ON"))
             {
-                text.fontMaterial.DisableKeyword("UNDERLAY_ON");
+                text.fontSharedMaterial.DisableKeyword("UNDERLAY_ON");
             }
 
             if (model.outlineWidth > 0f)
             {
-                text.fontMaterial.EnableKeyword("OUTLINE_ON");
+                text.fontSharedMaterial.EnableKeyword("OUTLINE_ON");
                 text.outlineWidth = model.outlineWidth;
                 text.outlineColor = model.outlineColor;
             }
-            else if (text.fontMaterial.IsKeywordEnabled("OUTLINE_ON"))
+            else if (text.fontSharedMaterial.IsKeywordEnabled("OUTLINE_ON"))
             {
-                text.fontMaterial.DisableKeyword("OUTLINE_ON");
+                text.fontSharedMaterial.DisableKeyword("OUTLINE_ON");
             }
```

Comment:
If you are already holding an instance of the material it would be better to use that directly
i.e: 
```
cachedFontMaterial.EnableKeyword("UNDERLAY_ON");
cachedFontMaterial.SetColor("_UnderlayColor", model.shadowColor);
cachedFontMaterial.SetFloat("_UnderlaySoftness", model.shadowBlur);
```

Reply:
We can't do it directly because the `ApplyModelChanges` function is static but not our `cachedFontMaterial` variable. We could pass the `cachedFontMaterial` variable through the function signature but, in order not to make the function more complex, I prefeer to continue using `text.fontSharedMaterial`.